src/ethereum/berlin/vm/instructions/bitwise.py

```diff
@@ -154,75 +180,91 @@ def get_byte(evm: Evm) -> None:
 
     push(evm.stack, result)
 
+    # PROGRAM COUNTER
     evm.pc += 1
 
 
 def bitwise_shl(evm: Evm) -> None:
     """
     Logical shift left (SHL) operation of the top 2 elements of the stack.
     Pushes the result back on the stack.
-
     Parameters
     ----------
     evm :
         The current EVM frame.
     """
-    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+    # STACK
     shift = pop(evm.stack)
     value = pop(evm.stack)
 
-    evm.pc += 1
-    shifted_value = 0
+    # GAS
+    charge_gas(evm, GAS_VERY_LOW)
+
+    # OPERATION
     if shift < 256:
-        shifted_value = (value << shift) % U256_CEIL_VALUE
+        result = U256((value << shift) % U256_CEIL_VALUE)
+    else:
+        result = U256(0)
+
+    push(evm.stack, result)
 
-    push(evm.stack, U256(shifted_value))
+    # PROGRAM COUNTER
+    evm.pc += 1
 
 
 def bitwise_shr(evm: Evm) -> None:
     """
     Logical shift right (SHR) operation of the top 2 elements of the stack.
     Pushes the result back on the stack.
-
     Parameters
     ----------
     evm :
         The current EVM frame.
     """
-    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+    # STACK
     shift = pop(evm.stack)
     value = pop(evm.stack)
 
-    evm.pc += 1
-    shifted_value = U256(0)
+    # GAS
+    charge_gas(evm, GAS_VERY_LOW)
+
+    # OPERATION
     if shift < 256:
-        shifted_value = value >> shift
+        result = value >> shift
+    else:
+        result = U256(0)
+
+    push(evm.stack, result)
 
-    push(evm.stack, shifted_value)
+    # PROGRAM COUNTER
+    evm.pc += 1
 
 
 def bitwise_sar(evm: Evm) -> None:
     """
     Arithmetic shift right (SAR) operation of the top 2 elements of the stack.
     Pushes the result back on the stack.
-
     Parameters
     ----------
     evm :
         The current EVM frame.
     """
-    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+    # STACK
     shift = pop(evm.stack)
-    value = pop(evm.stack)
+    signed_value = pop(evm.stack).to_signed()
 
-    signed_value = value.to_signed()
+    # GAS
+    charge_gas(evm, GAS_VERY_LOW)
 
-    evm.pc += 1
+    # OPERATION
     if shift < 256:
-        shifted_value = signed_value >> shift
+        result = U256.from_signed(signed_value >> shift)
     elif signed_value >= 0:
-        shifted_value = 0
+        result = U256(0)
     else:
-        shifted_value = U256.MAX_VALUE
+        result = U256(U256.MAX_VALUE)
```

Comment:
Ugh, does mypy not pick this up correctly? I thought `MAX_VALUE` was already a `U256`.